AirSim/AirLib/include/sensors/gps/GpsSimple.hpp

```diff
@@ -65,24 +68,55 @@ class GpsSimple : public GpsBase {
     //*** End: UpdatableState implementation ***//
 
     virtual ~GpsSimple() = default;
+    //generate Gaussian noise 
+    const float getGaussianNoise(float mean, float var)
+    {
+        std::normal_distribution<float> distribution(mean, var);
+        auto seed = (unsigned int)std::chrono::system_clock::now().time_since_epoch().count();
+        std::default_random_engine generator(seed);
+        return distribution(generator);
+    }
+    //Generate noisy geo_point for moving vehicle
+    msr::airlib::GeoPoint generateErrors(msr::airlib::GeoPoint geo_point, real_T eph, real_T epv)
+    {
+        HomeGeoPoint geo_point_in = HomeGeoPoint(geo_point);
+        float var_h = pow(eph, 2)/18;
+        float var_v = pow(epv, 2)/9;

```

Comment:
what do these 18 and 9 represent?

Reply:
Essentially they are a result of following the [3-sigma rule](https://en.wikipedia.org/wiki/68%E2%80%9395%E2%80%9399.7_rule) for Gaussian distribution so we are sure that the "random" noise we generate are within bounds.